src/ui/ui_element.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <vector>
+
+#include "render2d.h"
+
+class UIElement {
+public:
+    UIElement() {} // default ctor
+    virtual ~UIElement() {}
+    UIElement(float start_x, float start_y)
+        : start_x(start_x), start_y(start_y) {}
+    virtual void draw(Render2D &renderer_2d,
```

Comment:
Wait, sorry. How does this static dependency injection change the member variable? Or is ``const Render2D *renderer;`` also ``static``?